### DISCUSSION TITLE
Anybody thought of using `useSyncExternalStore` instead of React Context?

### DISCUSSION THREAD

**Discussion by swernerx** [ivanhofer/typesafe-i18n#605] [Ideas]
From what I understand in the code all locale loading merges data into one somewhat global data structure. Therefor [`useSyncExternalStore`](https://beta.reactjs.org/reference/react/useSyncExternalStore) which is a relatively new API in React would be a good match for accessing this data. Has anybody played around with that concept already?

**Comment by ivanhofer** [comment]
I'm not really involved in the react ecosystem anymore, so I'm always grateful if someone suggests an improvement.
Can you please summarize what the benefits and maybe downsides (only react version >18 I guess?) would be?
